android/app/src/main/java/updated/mysterium/vpn/ui/manual/connect/select/node/SelectNodeActivity.kt

```diff
@@ -1,91 +1,119 @@
+            val tabBinding = ItemTabBinding.bind(tab)
+            tabBinding.allNodesImageButton.text = resources.getString(TAB_ITEMS_CONTENT[index].textResId)
+            if (index == 0) {
+                tabBinding.allNodesImageButton.background = ContextCompat.getDrawable(
+                    this@SelectNodeActivity,
+                    (TAB_ITEMS_CONTENT[index].selectedBackgroundResId)
+                )
+            } else {
+                tabBinding.allNodesImageButton.background = ContextCompat.getDrawable(
+                    this@SelectNodeActivity,
+                    (TAB_ITEMS_CONTENT[index].unselectedBackgroundResId)
+                )
+            }
+            binding.chooseListTabLayout.getTabAt(index)?.customView = tab
         }
-    }
+        binding.chooseListTabLayout.addOnTabSelectedListener(
 
-    private fun bindsAction() {
-        binding.sortByView.setOnClickListener {
-            nodeViewModel.getSortedProposal().observe(
-                this,
-                { result ->
-                    result.onSuccess { proposalList ->
-                        countrySelectAdapter.replaceAll(proposalList)
-                    }
+            object : TabLayout.OnTabSelectedListener {
 
-                    result.onFailure { throwable ->
-                        Log.e(TAG, throwable.localizedMessage ?: throwable.toString())
-                        // TODO("Replace for error dialog or something else")
+                override fun onTabSelected(tab: TabLayout.Tab?) {
+                    tab?.customView?.let {
+                        ItemTabBinding.bind(it)
+                            .allNodesImageButton
+                            .background = ContextCompat.getDrawable(
+                            this@SelectNodeActivity,
+                            (TAB_ITEMS_CONTENT[tab.position].selectedBackgroundResId)
+                        )
                     }
                 }
-            )
-            nodeViewModel.changeSortType().observe(
-                this,
-                { result ->
-                    result.onSuccess {
-                        binding.sortByView.text = it.toString()
-                            .toLowerCase(Locale.getDefault())
-                            .capitalize(Locale.getDefault())
+
+                override fun onTabUnselected(tab: TabLayout.Tab?) {
+                    tab?.customView?.let {
+                        ItemTabBinding.bind(it)
+                            .allNodesImageButton
+                            .background = ContextCompat.getDrawable(
+                            this@SelectNodeActivity,
+                            (TAB_ITEMS_CONTENT[tab.position].unselectedBackgroundResId)
+                        )
+
                     }
                 }
-            )
-        }
+
+                override fun onTabReselected(tab: TabLayout.Tab?) {
+                    // empty
+                }
```

Comment:
you can make custom util that inherits that listener, and override particular function with no implementation without needed implementation there, so you will have no empty functions in your view class

Reply:
Fix